plugins/VersionUpgrade/VersionUpgrade44to45/VersionUpgrade44to45.py

```diff
@@ -75,8 +93,42 @@ def upgradeStack(self, serialized: str, filename: str) -> Tuple[List[str], List[
         # Update version number.
         if "metadata" not in parser:
             parser["metadata"] = {}
+
+        # Check for leftover hidden machines. They should be deleted along with their associated files
+        stack_id = parser["general"]["id"]
+        if parser["metadata"]["setting_version"] == "10" and stack_id is not None:
+            stack_type = parser["metadata"]["type"]
+            if stack_type == "machine" and parser["metadata"]["hidden"]:
+                if not self._hidden_machines:
+                    # schedule a final upgrade step to delete all hidden machines
+                    VersionUpgradeManager.getInstance().upgradeExtraTask(CallableTask(self.deleteHiddenMachines))
+                self._hidden_machines[stack_id] = filename
+            elif stack_type == "extruder_train":
+                machine = parser["metadata"]["machine"]
+                self._extruder_machine_mapping.append(ExtruderMachineMapping(stack_id, filename, machine))
+
         parser["metadata"]["setting_version"] = "11"
 
         result = io.StringIO()
         parser.write(result)
         return [filename], [result.getvalue()]
+
+    def deleteHiddenMachines(self) -> None:
+
+        # Delete extruders associated to hidden machines
+        for item in self._extruder_machine_mapping:
+            if item.machine_id in self._hidden_machines:
+                Logger.debug("Deleting extruder files for {}".format(item))
+                self._deleteConfigFiles(item.extruder_filename)
```

Comment:
This is not going to work in the future any more if any of our future upgrade tasks decide to rename the files.